### PR TITLE
Create a new fixture for creating scale pods and PVCs using a Kube job on MS consumers

### PR DESCRIPTION
**Pull request by yitzhak12**

### CODE REVIEW

File: ocs_ci/ocs/scale_lib.py [hunk 1]
```diff
@@ -108,6 +110,8 @@ def create_multi_pvc_pod(
             io_runtime (seconds): Runtime in Seconds to continue IO
             pvc_size (int): Size of PVC to be created
             max_pvc_size (int): The max size of the pvc
+            remove_security_context_section (bool): If True, remove the security section from the Yaml file.
```
Comment (ramkiperiy) [inline ocs_ci/ocs/scale_lib.py:113]: let's keep it default as True, i.e. to remove security context section in yaml.

File: ocs_ci/ocs/scale_lib.py [hunk 2]
```diff
@@ -95,6 +96,7 @@ def create_multi_pvc_pod(
         io_runtime=None,
         pvc_size=None,
         max_pvc_size=105,
+        remove_security_context_section=False,
```
Comment (ramkiperiy) [inline ocs_ci/ocs/scale_lib.py:99]: Does false mean we are not removing it by default? let's make it default to remove the content in yaml.

File: ocs_ci/ocs/scale_lib.py [hunk 3]
```diff
@@ -1361,6 +1375,7 @@ def attach_multiple_pvc_to_pod_dict(
     io_runtime=None,
     io_size=None,
     pod_yaml=constants.PERF_POD_YAML,
+    remove_security_context_section=False,
```
Comment (ramkiperiy) [inline ocs_ci/ocs/scale_lib.py:1378]: same here, let's keep this value as True by default.

File: tests/conftest.py [hunk 4]
```diff
@@ -5353,15 +5353,17 @@ def create_scale_pods_and_pvcs_using_kube_job(request):
...
 
     def factory(
         scale_count=None,
-        pvc_per_pod_count=10,
+        pvc_per_pod_count=5,
         start_io=True,
         io_runtime=None,
         pvc_size=None,
         max_pvc_size=30,
+        remove_security_context_section=False,
```
Comment (ramkiperiy) [inline tests/conftest.py:5366]: same here.

File: ocs_ci/ocs/scale_lib.py (hunk 2 above)
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py:99]: Yes. If it is False, that means that we are not removing it. Are you sure removing it from the `performance.yaml` file is ok? from my side, changing the default value to True is ok.

File: ocs_ci/ocs/scale_lib.py (hunk 1 above)
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py:113]: Okay, np.

File: ocs_ci/ocs/scale_lib.py (hunk 3 above)
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py:1378]: yes, np

File: tests/conftest.py (hunk 4 above)
Comment (yitzhak12) [reply tests/conftest.py:5366]: Okay

File: ocs_ci/ocs/scale_lib.py [hunk 5]
```diff
@@ -1390,6 +1407,8 @@ def attach_multiple_pvc_to_pod_dict(
         if len(temp_list) == pvcs_per_pod:
             pod_dict = pod_yaml
             pod_data = templating.load_yaml(pod_dict)
+            if remove_security_context_section:
+                del pod_data["spec"]["containers"][0]["securityContext"]
```
Comment (PrasadDesala) [inline ocs_ci/ocs/scale_lib.py:1411]: This is not needed, the security context section is removed from the performance yaml very recently.

File: ocs_ci/ocs/scale_lib.py [hunk 6]
```diff
@@ -108,6 +110,8 @@ def create_multi_pvc_pod(
             io_runtime (seconds): Runtime in Seconds to continue IO
             pvc_size (int): Size of PVC to be created
             max_pvc_size (int): The max size of the pvc
+            remove_security_context_section (bool): If True, remove the security context section from the Yaml file.
+                False, otherwise. Default value is True.
```
Comment (PrasadDesala) [inline ocs_ci/ocs/scale_lib.py:114]: You can remove all the code changes in this PR related to security_context.
Also rebase the PR if not done recently to pull the latest changes which has the perf yaml changes

File: ocs_ci/ocs/scale_lib.py (hunk 5 above)
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py]: Oh, Okay. I will remove it.

**Comment by yitzhak12** [comment]
The pr validation passed with two consumers and a provider: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/407/console.

**Review by OdedViner — APPROVED** [review]

**Pull request merged**